stdlib/slack/slack_test.go

```diff
@@ -67,8 +68,16 @@ func NewServer(t *testing.T) *Server {
 			URL:           r.URL.String(), // r.URL.String(),
 			Authorization: r.Header.Get("Authorization"),
 		}
-		dec := json.NewDecoder(r.Body)
-		err := dec.Decode(&sr.PostData)
+		b, err := ioutil.ReadAll(r.Body)
+		defer r.Body.Close()
+    if err != nil {
+			t.Error(err)
+		}
+		err = json.Unmarshal(b,&sr.PostData)
+		if err != nil {
+			t.Error(err)
+		}
+		err = json.Unmarshal(b,&sr.RawData)
```

Comment:
This is necessary because I need the test to differentiate between JSON with containing `"as_user":null` and JSON that does not contain an `as_user` field whatsoever; the former will be rejected by Slack's API.

Unfortunately, as far as I am aware, there is no way to distinguish those two JSON documents once the JSON has been decoded into a Go struct; so, for the extra test, I need it decoded as a `map[string]interface{}`, where the distinction between a key being absent vs being null is preserved.

Reply:
Could you summarize that as a comment in the patch itself? Maybe it's not a huge deal, but I'll definitely forget that, and won't quite understand why we're testing what we're testing.

Reply:
Sure, I can do that

Reply:
I added the requested comment, though I added it to the Struct declaration where I define the `RawData` map, rather htan the lines where I populate it. I can move it if you'd prefer it go elsewhere.